### PR TITLE
DOB hash calculation & wiring (EXPOSUREAPP-7488, EXPOSUREAPP-7509)

### PR DESCRIPTION
**Pull request by d4rken**
* Adds date of birth hash calculation for PCR test registration
* Wires up date of birth hash to be passed from to verification server (UI test registration request -> corona test repo -> test processor -> playbook -> server -> api)
* Extends verification server API with new attribute and adjusts request padding calculation

### CODE REVIEW

**Review by mtwalli — DISMISSED** [review]

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/server/VerificationApiV1.kt [hunk 1]
```diff
@@ -8,9 +8,10 @@ import retrofit2.http.POST
 interface VerificationApiV1 {
 
     data class RegistrationTokenRequest(
-        @SerializedName("keyType") val keyType: String? = null,
-        @SerializedName("key") val key: String? = null,
-        @SerializedName("requestPadding") val requestPadding: String? = null
+        @SerializedName("keyType") val keyType: VerificationKeyType,
+        @SerializedName("key") val key: String,
+        @SerializedName("keyDob") val dateOfBirthKey: String? = null,
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/server/VerificationApiV1.kt:13]: The new optional attribute

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/server/VerificationKeyType.kt [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
 
+import com.google.gson.annotations.SerializedName
+
 enum class VerificationKeyType {
-    GUID, TELETAN;
+    @SerializedName("GUID")
+    GUID,
+
+    @SerializedName("TELETAN")
+    TELETAN;
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/server/VerificationKeyType.kt:10]: We previously just relied on the enums never being renamed or obfuscated, this was a bit too fragile for my taste.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/server/VerificationServer.kt [hunk 3]
```diff
@@ -86,41 +152,72 @@ class VerificationServer @Inject constructor(
...
+        // `cwa-fake 0\n` -> 12
+        private const val HEADER_SIZE_VAL_FAKE = 12
 
-        // padding test result
-        const val PADDING_LENGTH_HEADER_TEST_RESULT = 7
-        const val PADDING_LENGTH_BODY_TEST_RESULT = 31 + VERIFICATION_BODY_FILL
+        // `cwa-header-padding\n` -> 22
+        private const val HEADER_SIZE_VAL_PADDING = 22
+        private const val HEADER_SIZE_OVERHEAD = HEADER_SIZE_VAL_FAKE + HEADER_SIZE_VAL_PADDING
+        private const val HEADER_SIZE_OUR_DATA = HEADER_SIZE_LONGEST_METHOD + HEADER_SIZE_OVERHEAD
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/server/VerificationServer.kt:212]: We pad the request for plausible deniability, so they all have the same size.
Previously the padding was difficult to understand why a certain value was picked, as we summed up the differences to other API requests, this was a bit too :mage:.
This PR changes the padding calculation to start of with an expected size, and then each API call expects its own size, and what we are left with is the  required padding.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/type/common/DateOfBirthKey.kt [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    init {
+        require(testGuid.isNotEmpty()) { "GUID can't be empty." }
+    }
+
+    val key by lazy {
+        val dobFormatted = dateOfBirth.toString(DOB_FORMATTER)
+        val keyHash = "${testGuid}$dobFormatted".toSHA256()
+        "x${keyHash.substring(1)}"
+    }
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/type/common/DateOfBirthKey.kt:20]: Date of Birth hash calculation

**Review by kolyaopahle — APPROVED** [review]
Love the new padding calculations ❤️  tests seem good to me aswell

**Review by chris-cwa — APPROVED** [review]

**Pull request merged**